src/server/database_provider.py

```diff
@@ -0,0 +1,24 @@
+import sqlite3
+import os
+
+from constants import PATHS
```

Comment:
I suggest adding `PATHS` class to `__init__.py` file in this module with:
```python
from .constants import PATHS
```

and here change it to:
```python
from . import PATHS
```

The reason behind is to take advantage of python module mechanic, avoiding eventual cyclic imports.